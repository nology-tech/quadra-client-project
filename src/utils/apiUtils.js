export const getCurrency = async (currency) => {
  const url =
    `https://money-app-api-oi3xuie5la-nw.a.run.app/quadra-money/v1/currencies/${currency}`;
  const res = await fetch(url);
  const data = await res.json();
  return data.rates;
};

export const getCurrencyGBP = async () => {
  const url =
    "https://money-app-api-oi3xuie5la-nw.a.run.app/quadra-money/v1/currencies/gbp";
  const res = await fetch(url);
  const data = await res.json();
  return data.rates;
};

export const getUserContacts = async () => {
  const url = "https://money-app-api-oi3xuie5la-nw.a.run.app//quadra-money/v1/contacts";
    const res = await fetch(url);
    const data = await res.json();
    return data;
}

export const storeUserDetails = async (userID, name) => {
  const url= "https://money-app-api-oi3xuie5la-nw.a.run.app/quadra-money/v1/user";
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Accept': 'application/json',
      'Content-Type': 'application/json', },
    body: JSON.stringify({ 
      userID: userID,
      firstName: name,
      lastName: '',
      middleNames: '',
  })
  });
  const json = await res.json();
  return json;
}

export const storeUserHoldings = async (userID, currencyName, amount, currencyCode, currencySymbol) => {
  const url= "https://money-app-api-oi3xuie5la-nw.a.run.app//quadra-money/v1/holding";
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Accept': 'application/json',
      'Content-Type': 'application/json', },
    body: JSON.stringify({ 
      userID: userID,
      currencyName: currencyName,
      amount: amount ,
      currencyCode: currencyCode,
      currencySymbol: currencySymbol,
  })
  });
  const json = await res.json();
  return json;
}

export const getUserHoldings = async (userID) => {
  const url= `https://money-app-api-oi3xuie5la-nw.a.run.app//quadra-money/v1/user-holding/${userID}`;
  const res = await fetch(url);
  const json = await res.json();
  return json;
}