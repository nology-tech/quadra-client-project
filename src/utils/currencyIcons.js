import Flags from "country-flag-icons/react/3x2";

const iconMap = {
    "GBP": <Flags.GB className="flag" />,
    "USD": <Flags.US className="flag" />,
    "AUD": <Flags.AU className="flag" />,
    "ZAR": <Flags.ZA className="flag" />,
    "JPY": <Flags.JP className="flag" />,
    "EUR": <Flags.EU className="flag" />
};

export default iconMap;