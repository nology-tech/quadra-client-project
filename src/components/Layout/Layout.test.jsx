import { customRender } from "../../utils/testUtils";
import Layout from "./Layout";

it("Should render layout with children", () => {
  const { container } = customRender(
    <Layout>
      <p>some content</p>
    </Layout>
  );

  expect(container).toMatchSnapshot();
});

it("Should render without navbar when showNavbar is false", () => {
  const { container } = customRender(
    <Layout showNavbar={false}>
      <p>some content</p>
    </Layout>
  );

  expect(container).toMatchSnapshot();
});
