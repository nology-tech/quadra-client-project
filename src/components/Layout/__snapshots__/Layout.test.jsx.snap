// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render layout with children 1`] = `
<div>
  <div
    class="layout"
  >
    <p>
      some content
    </p>
  </div>
</div>
`;

exports[`Should render without navbar when showNavbar is false 1`] = `
<div>
  <div
    class="layout"
  >
    <p>
      some content
    </p>
  </div>
</div>
`;
